pkg/prometheus/statefulset.go

```diff
@@ -97,153 +71,6 @@ func prometheusNameByShard(name string, shard int32) string {
 	return fmt.Sprintf("%s-shard-%d", base, shard)
 }
 
-func makeStatefulSet(
-	logger log.Logger,
-	name string,
-	p monitoringv1.Prometheus,
-	config *operator.Config,
-	ruleConfigMapNames []string,
-	inputHash string,
-	shard int32,
-	tlsAssetSecrets []string,
-) (*appsv1.StatefulSet, error) {
-	// p is passed in by value, not by reference. But p contains references like
-	// to annotation map, that do not get copied on function invocation. Ensure to
-	// prevent side effects before editing p by creating a deep copy. For more
-	// details see https://github.com/prometheus-operator/prometheus-operator/issues/1659.
-	p = *p.DeepCopy()
```

Comment:
IIUC the operator already calls DeepCopy() at the beginning of sync() so there's probably no reason to clone the object again.

https://github.com/prometheus-operator/prometheus-operator/blob/363a4a7dfcd1e20c3a4737a6ba3f45e03a6d702b/pkg/prometheus/operator.go#L1378-L1382

Reply:
addressed in #4950